dashboard/lib/sample_data.rb

```diff
@@ -0,0 +1,199 @@
+
+class SampleData
+  SAMPLE_TEACHER_EMAIL = 'testteacher@code.org'.freeze
+  SAMPLE_TEACHER_PASSWORD = '00secret'.freeze
+  SAMPLE_TEACHER_NAME = 'TestTeacher Codeberg'.freeze
+  SAMPLE_STUDENT_NAME_FORMAT = 'TestStudent%s Codeberg'.freeze
+  SAMPLE_STUDENT_NAME_REGEX = /TestStudent\d* Codeberg/
+
+  @@rng = nil
+
+  # Returns a seeded random number generator for consistent test data
+  def self.rng
+    @@rng ||= Random.new(0)
+  end
+
+  # Creates sample data
+  def self.seed
+    raise "Should not be run outside of development" unless CDO.rack_env?(:development)
+
+    # Create a test teacher
+    teacher = create_teacher SAMPLE_TEACHER_EMAIL, SAMPLE_TEACHER_PASSWORD,
+      SAMPLE_TEACHER_NAME
+
+    # Create normal-sized and large sections for each of CSF, CSD and CSP
+
+    create_section(
+      teacher: teacher, name: 'CSF 1',
+      login_type: Section::LOGIN_TYPE_PICTURE, grade: 2, age_min: 7,
+      age_max_inclusive: 9, script_name: 'coursea-2018', num_students: 30,
+      use_imperfect_results: true
+    )
+
+    create_section(
+      teacher: teacher, name: 'CSF 2 (Large)',
+      login_type: Section::LOGIN_TYPE_PICTURE, grade: 2, age_min: 7,
+      age_max_inclusive: 9, script_name: 'coursea-2018', num_students: 150,
+      use_imperfect_results: true
+    )
+
+    create_section(
+      teacher: teacher, name: 'CSD 1',
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 7, age_min: 13,
+      age_max_inclusive: 14, script_name: 'csd1-2018', num_students: 30,
+      use_imperfect_results: false
+    )
+
+    create_section(
+      teacher: teacher, name: 'CSD 2 (Large)',
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 7, age_min: 13,
+      age_max_inclusive: 14, script_name: 'csd1-2018', num_students: 150,
+      use_imperfect_results: false
+    )
+
+    create_section(
+      teacher: teacher, name: 'CSP 1',
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 10, age_min: 14,
+      age_max_inclusive: 16, script_name: 'csp1-2018', num_students: 30,
+      use_imperfect_results: false
+    )
+
+    create_section(
+      teacher: teacher, name: 'CSP 2 (Large)',
+      login_type: Section::LOGIN_TYPE_EMAIL, grade: 10, age_min: 14,
+      age_max_inclusive: 16, script_name: 'csp1-2018', num_students: 150,
+      use_imperfect_results: false
+    )
+  end
+
+  # Hard-delete the teacher and all of the teacher's sections and students
+  # and recreate. Sections and followers would be soft-deleted by
+  # dependency when we delete the teacher; but to not leave a trail of
+  # old test data behind, we explictly hard-delete.
+  def self.create_teacher(email, password, name)
+    # Delete any existing test data
+    user = User.find_by_email_or_hashed_email(email)
+    unless user.nil?
+      user.sections.each do |section|
+        # Hard-delete all students in each section.
+        section.students.each do |student_user|
+          raise "Not a sample student" unless student_user.name =~ SAMPLE_STUDENT_NAME_REGEX
```

Comment:
Probably worth adding some identifiable information in the error, like student id or email.

Reply:
Thx, done